### PR TITLE
Add shutdown duration option to Node.js simple 

### PR DESCRIPTION
**Pull request by steven-supersolid**
The example code now reads the first argument if specified
If --help is passed then will exit with a help message
If a number is passed then that will be used for the shutdown duration

Converted setTimeout to use `await` to make code clearer
Updated watch to give more relevant information set during the test
Clear intervals correctly so these don't prevent the app from exiting
Found the app still won't exit cleanly so potentially a bug in `sdk.close()` (fix as separate PR)

Updated Makefile and dockerfile to pass in a parameter
Updated documentation

### CODE REVIEW

File: examples/nodejs-simple/Makefile [hunk 1]
```diff
@@ -43,7 +43,7 @@ build:
 	cd $(root_path) && docker build -f $(project_path)/Dockerfile --tag=$(server_tag) .
 
 run:
-	docker run --network=host $(server_tag)
+	docker run --network=host $(server_tag) $(args)
```
Comment (markmandel) [inline examples/nodejs-simple/Makefile]: Nice stuff - will be useful for testing #1299 :+1: 

We should also increment the tag version in this file, so we can rebuild and push it up for people to use.

File: examples/nodejs-simple/README.md [hunk 2]
```diff
@@ -2,11 +2,14 @@
 
 This is a very simple "server" that doesn't do much other than show how the SDK works in node.js.
 
-It will
-- Setup the Agones SDK
-- Call `sdk.ready()` to register that it is ready with Agones.
+It will:
```
Comment (markmandel) [inline examples/nodejs-simple/README.md]: Do we want to document what the command is (or the Docker run command?) should be to run this for longer than 60 seconds?

File: examples/nodejs-simple/Makefile (hunk 1 above)
Comment (steven-supersolid) [reply examples/nodejs-simple/Makefile]: Would that be server-tag and should it be 1.5 now?

File: examples/nodejs-simple/README.md (hunk 2 above)
Comment (steven-supersolid) [reply examples/nodejs-simple/README.md]: That is documented around line 120 after example debug output but perhaps clearer if I reorder?

File: examples/nodejs-simple/package.json [hunk 3]
```diff
@@ -1,6 +1,6 @@
 {
   "dependencies": {
-    "@google-cloud/agones-sdk": "1.3.0"
+    "@google-cloud/agones-sdk": "1.4.0"
```
Comment (steven-supersolid) [inline examples/nodejs-simple/package.json]: Perhaps should update as part of the release @markmandel ?

Comment (markmandel) [reply examples/nodejs-simple/package.json]: I don't think this is necessary, unless there is a code change in either the SDK or the example.

Unless you can see a good reason to?

File: examples/nodejs-simple/Makefile (hunk 1 above)
Comment (markmandel) [reply examples/nodejs-simple/Makefile]: `server_tag = $(REPOSITORY)/nodejs-simple-server:0.3` :arrow_right: `server_tag = $(REPOSITORY)/nodejs-simple-server:0.4`

File: examples/nodejs-simple/README.md [hunk 4]
```diff
@@ -39,71 +42,98 @@ docker run --network=host gcr.io/agones-images/nodejs-simple-server:0.1
...
+You can optionally specify how long the server will stay up once the basic tests are complete.
+To do this pass arguments through, e.g. to increase the shutdown duration to 120 seconds:
+```
+$ make args="120" run
+```
+
+Please note that there is a max sleep time in Node.js of 2,147,483 seconds (~24 days). Values will be capped at this and negative values or zero will also be set to this.
+So to keep the server alive for a long time you could use the following:
+```
+$ make args="0" run
```
Comment (markmandel) [inline examples/nodejs-simple/README.md]: I would make these (no pun intended :smile: ) the full Docker run command.

That way people don't need to pull down the full example source code -- and can also replicated within their Agones install as well.

File: examples/nodejs-simple/README.md (hunk 2 above)
Comment (markmandel) [reply examples/nodejs-simple/README.md]: I would also like the explicit command line argument documented.

Or is it just "120" as seconds ? Should we use a key-value pair? Like --timeout=120 ?

Would be much more explicit, and also expandable down the line. WDYT?

File: examples/nodejs-simple/package.json (hunk 3 above)
Comment (steven-supersolid) [reply examples/nodejs-simple/package.json]: The main reason would be to keep the code sample up to date and consistent with the release. We could also change to a non-pinned dependency

File: examples/nodejs-simple/README.md (hunk 4 above)
Comment (steven-supersolid) [reply examples/nodejs-simple/README.md]: Ah sure. I was just basing this on the previous instructions. I usually run with `npm start` anyway.
If there's a need for both should we have both ways of running?
@roberthbailey you wrote the original README - any thoughts on this?

File: examples/nodejs-simple/package.json (hunk 3 above)
Comment (markmandel) [reply examples/nodejs-simple/package.json]: But if there is no public API code change between versions, then it is up to date with the latest release :smile:

File: examples/nodejs-simple/README.md (hunk 4 above)
Comment (markmandel) [reply examples/nodejs-simple/README.md]: Ah, if it's been this way for a while, I'll be less opinionated on changing to Docker run... but it still would be nice I think?

File: examples/nodejs-simple/README.md [hunk 5]
```diff
@@ -32,78 +36,44 @@ $ make build
...
+```
+$ make args="--help" run
+$ docker run --network=host gcr.io/agones-images/nodejs-simple-server:0.4 --help
+$ npm start -- --help
+```
+
+You can optionally specify how long the server will stay up once the basic tests are complete with the `--timeout` option.
+To do this pass arguments through, e.g. to increase the shutdown duration to 120 seconds:
+```
+$ make args="--timeout=120" run
```
Comment (markmandel) [inline examples/nodejs-simple/README.md]: Love this! :smile:

**Comment by markmandel** [comment]
Weird, I can't update this branch to master. There is no button? 
image

**Comment by steven-supersolid** [comment]
Wonder if it's because I merged without the button last time? But I had the button just now, so pressed it :)

**Comment by markmandel** [comment]
I'll wait for this to run through, and then it gets to be first in the queue :smile:

**Pull request merged**